dlt/destinations/impl/sqlalchemy/db_api_client.py

```diff
@@ -0,0 +1,378 @@
+            self.execute_sql(sa.schema.DropSchema(self.dataset_name))
+
+    def truncate_tables(self, *tables: str) -> None:
+        # TODO: alchemy doesn't have a construct for TRUNCATE TABLE
+        for table in tables:
+            tbl = sa.Table(table, self.metadata, schema=self.dataset_name, keep_existing=True)
+            self.execute_sql(tbl.delete())
+
+    def drop_tables(self, *tables: str) -> None:
+        for table in tables:
+            tbl = sa.Table(table, self.metadata, schema=self.dataset_name, keep_existing=True)
+            self.execute_sql(sa.schema.DropTable(tbl, if_exists=True))
+
+    def execute_sql(
+        self, sql: Union[AnyStr, sa.sql.Executable], *args: Any, **kwargs: Any
+    ) -> Optional[Sequence[Sequence[Any]]]:
+        with self.execute_query(sql, *args, **kwargs) as cursor:
+            if cursor.returns_rows:
+                return cursor.fetchall()
+            return None
+
+    @contextmanager
+    def execute_query(
+        self, query: Union[AnyStr, sa.sql.Executable], *args: Any, **kwargs: Any
+    ) -> Iterator[DBApiCursor]:
+        if isinstance(query, str):
+            if args:
+                # Sqlalchemy text supports :named paramstyle for all dialects
+                query, kwargs = self._to_named_paramstyle(query, args)  # type: ignore[assignment]
+                args = ()
+            query = sa.text(query)
+        with self._transaction():
+            yield SqlaDbApiCursor(self._current_connection.execute(query, *args, **kwargs))  # type: ignore[abstract]
+
+    def get_existing_table(self, table_name: str) -> Optional[sa.Table]:
+        """Get a table object from metadata if it exists"""
+        key = self.dataset_name + "." + table_name
+        return self.metadata.tables.get(key)  # type: ignore[no-any-return]
+
+    def create_table(self, table_obj: sa.Table) -> None:
+        table_obj.create(self._current_connection)
+
+    def _make_qualified_table_name(self, table: sa.Table, escape: bool = True) -> str:
+        if escape:
+            return self.dialect.identifier_preparer.format_table(table)  # type: ignore[attr-defined,no-any-return]
+        return table.fullname  # type: ignore[no-any-return]
+
+    def make_qualified_table_name(self, table_name: str, escape: bool = True) -> str:
+        tbl = self.get_existing_table(table_name)
+        if tbl is None:
+            tmp_metadata = sa.MetaData()
+            tbl = sa.Table(table_name, tmp_metadata, schema=self.dataset_name)
+        return self._make_qualified_table_name(tbl, escape)
+
+    def alter_table_add_column(self, column: sa.Column) -> None:
+        """Execute an ALTER TABLE ... ADD COLUMN ... statement for the given column.
+        The column must be fully defined and attached to a table.
+        """
+        # TODO: May need capability to override ALTER TABLE statement for different dialects
+        alter_tmpl = "ALTER TABLE {table} ADD COLUMN {column};"
```

Comment:
there's no ALTER TABLE in SQLAlchemy? we are reflecting all tables before update schema. maybe we could reuse those tables to add columns and sync?

Reply:
There isn't unfortunately anything I could find, I dug through all the DDL generation stuff. I was planning to experiment with bringing in alembic for this if it's not too complicated, that should work at least with the built-in dialects.   

Sqlalchemy itself only handles creating tables, `Table().create()`, etc, that doesn't alter any existing tables.

Reply:
OMG! I see